modules/gapi/src/backends/cpu/gparsers.cpp

```diff
@@ -0,0 +1,288 @@
+        int obj_index = index(i, b, m_lcoords);
+        return m_out[obj_index];
+    }
+
+    double x(int i, int b) {
+        int box_index = index(i, b, 0);
+        int col = i % m_side;
+        return (col + m_out[box_index]) / m_side;
+    }
+
+    double y(int i, int b) {
+        int box_index = index(i, b, 0);
+        int row = i / m_side;
+        return (row + m_out[box_index + m_side*m_side]) / m_side;
+    }
+
+    double width(int i, int b, float anchor) {
+        int box_index = index(i, b, 0);
+        return std::exp(m_out[box_index + 2*m_side*m_side]) * anchor / m_side;
+    }
+
+    double height(int i, int b, float anchor) {
+        int box_index = index(i, b, 0);
+        return std::exp(m_out[box_index + 3*m_side*m_side]) * anchor / m_side;
+    }
+
+    float classConf(int i, int b, int label) {
+         int class_index = index(i, b, m_lcoords + 1 + label);
+         return m_out[class_index];
+    }
+
+    cv::Rect toBox(double x, double y, double h, double w, cv::Size in_sz) {
+        auto h_scale = in_sz.height;
+        auto w_scale = in_sz.width;
+        cv::Rect r;
+        r.x = static_cast<int>((x - w / 2) * w_scale);
+        r.y = static_cast<int>((y - h / 2) * h_scale);
+        r.width = static_cast<int>(w * w_scale);
+        r.height = static_cast<int>(h * h_scale);
+        return r;
+    }
+
+private:
+    const float* m_out;
+    int m_side, m_lcoords, m_lclasses;
+
+    int index(int i, int b, int entry) {
+        return b * m_side * m_side * (m_lcoords + m_lclasses + 1) + entry * m_side * m_side + i;
+    }
+};
+
+struct YoloParams {
+    int num = 5;
+    int coords = 4;
+};
+
+class SSDParser
+{
+public:
+    SSDParser(const cv::MatSize& in_ssd_dims, const cv::Size& in_size, const float* const data)
+        : m_dims(in_ssd_dims), m_maxProp(in_ssd_dims[2]), m_objSize(in_ssd_dims[3]),
+          m_data(data), m_surface(cv::Rect({0,0}, in_size)), m_size(in_size) {
+        GAPI_Assert(in_ssd_dims.dims() == 4u); // Fixed output layout
+        GAPI_Assert(m_objSize  == 7);          // Fixed SSD object size
+    }
+
+    void adjustBoundingBox(cv::Rect& boundingBox) {
+        auto w = boundingBox.width;
+        auto h = boundingBox.height;
+
+        boundingBox.x -= static_cast<int>(0.067 * w);
+        boundingBox.y -= static_cast<int>(0.028 * h);
+
+        boundingBox.width += static_cast<int>(0.15 * w);
+        boundingBox.height += static_cast<int>(0.13 * h);
+
+        if (boundingBox.width < boundingBox.height) {
+            auto dx = (boundingBox.height - boundingBox.width);
+            boundingBox.x -= dx / 2;
+            boundingBox.width += dx;
+        } else {
+            auto dy = (boundingBox.width - boundingBox.height);
+            boundingBox.y -= dy / 2;
+            boundingBox.height += dy;
+        }
+    }
+
+    std::tuple<cv::Rect, float, float, float> unitedParse(const size_t& step) {
+        const float* it = m_data + step * m_objSize;
```

Comment:
should it be const float* const as in the constructor?

Reply:
Yes. I don't want change `it`.

Reply:
Decided to have const pointer* everywhere